.github/scripts/branch_pr_issue_closer.py

```diff
@@ -135,176 +98,162 @@ def _main_prog():
     #Log-in to github API using token
     #++++++++++++++++++++++++++++++++
 
-    ghub = Github(token)
+    auth = Auth.Token(token)
+    ghub = Github(auth=auth)
 
-    #+++++++++++++++++++++
+    #+++++++++++++++++++++++++
     #Open ESCOMP/CAM-SIMA repo
-    #+++++++++++++++++++++
+    #+++++++++++++++++++++++++
 
     cam_repo = ghub.get_repo("ESCOMP/CAM-SIMA")
 
-    #+++++++++++++++++++++++++++++
-    #Get triggering commit message
-    #+++++++++++++++++++++++++++++
+    #++++++++++++++++++++++++++++++
+    #Get PRs associated with commit
+    #++++++++++++++++++++++++++++++
 
     github_commit = cam_repo.get_commit(trigger_sha)
 
-    commit_message = github_commit.commit.message
+    commit_prs = github_commit.get_pulls()
 
-    #+++++++++++++++++++++++++++++++
-    #Search for github PR merge text
-    #+++++++++++++++++++++++++++++++
+    pr_nums = [pr.number for pr in commit_prs]
 
-    #Compile Pull Request merge text expression:
-    pr_merge_pattern = re.compile(r'Merge pull request ')
 
-    #Search for merge text, starting at beginning of message:
-    commit_msg_match = pr_merge_pattern.match(commit_message)
+    #If list is empty, then no PRs are associated
+    #with this commit, so go ahead and close:
+    if not pr_nums:
+        endmsg = f"No PRs associated with commit:\n{trigger_sha}\n"
+        endmsg += " so issue-closing script is stopping here."
+        end_script(endmsg)
 
-    #Check if match exists:
-    if commit_msg_match is not None:
-        #If it does then pull out text immediately after message:
-        post_msg_text = commit_message[commit_msg_match.end():]
+    #++++++++++++++++++++++++++++
+    #Loop over all associated PRs
+    #++++++++++++++++++++++++++++
 
-        #Split text into individual words:
-        post_msg_word_list = post_msg_text.split()
+    for pr_num in pr_nums:
 
-        #Extract first word:
-        first_word = post_msg_word_list[0]
+        #+++++++++++++++++++++++++++++++++++++
+        #Check that PR has in fact been merged
+        #+++++++++++++++++++++++++++++++++++++
 
-        #Print merged pr number to screen:
-        print(f"Merged PR: {first_word}")
+        #Extract pull request info:
+        merged_pull = cam_repo.get_pull(pr_num)
 
-        try:
-            #Try assuming the word is just a number:
-            pr_num = int(first_word[1:]) #ignore "#" symbol
-        except ValueError:
-            #If the conversion fails, then this is likely not a real PR merge, so end the script:
-            endmsg = "No Pull Request number was found in the commit message, so there is nothing for the script to do."
+        #If pull request has not been merged, then exit script:
+        if not merged_pull.merged:
+            endmsg = f"Pull request associated with commit:\n{trigger_sha}\n"
```

Comment:
Might be nice to include PR number
```suggestion
            endmsg = f"Pull request #{pr_num} associated with commit:\n{trigger_sha}\n"
```

Reply:
Yes!  PR number added.